### ISSUE TITLE
magento 2 does not support composer 2.x

### ISSUE THREAD

**Issue by nwtben** [graycoreio/mage2docker#15]
# :bug: Bug report

## Current Behavior
`docker-compose exec magento2 composer install` exits with return code 2. Log: 
```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires magento/composer-root-update-plugin ~1.0 -> satisfiable by magento/composer-root-update-plugin[1.0.0].
    - magento/composer-root-update-plugin 1.0.0 requires composer-plugin-api ^1.0 -> found composer-plugin-api[2.0.0] but it does not match the constraint.
  Problem 2
    - Root composer.json requires dealerdirect/phpcodesniffer-composer-installer ^0.5.0 -> satisfiable by dealerdirect/phpcodesniffer-composer-installer[v0.5.0].
    - dealerdirect/phpcodesniffer-composer-installer v0.5.0 requires composer-plugin-api ^1.0 -> found composer-plugin-api[2.0.0] but it does not match the constraint.
  Problem 3
    - magento/magento-composer-installer[0.1.11, ..., 0.1.13] require composer-plugin-api ^1.0 -> found composer-plugin-api[2.0.0] but it does not match the constraint.
    - magento/product-community-edition 2.4.1 requires magento/magento-composer-installer >=0.1.11 -> satisfiable by magento/magento-composer-installer[0.1.11, 0.1.12, 0.1.13].
    - Root composer.json requires magento/product-community-edition 2.4.1 -> satisfiable by magento/product-community-edition[2.4.1].

You are using Composer 2, which some of your plugins seem to be incompatible with. Make sure you update your plugins or report a plugin-issue to ask them to support Composer 2.
```

## Expected Behavior
`docker-compose exec magento2 composer install` should be able to resolve the latest requirements by magento/magento2.git in to a installable set of packages.

## Minimal reproduction of the problem with instructions
Getting Started with an existing Magento application but without a composer.lock file

## What is the motivation / use case for changing the behavior?
building composer.lock in a CI env

## Environment
tested in ubuntu 18.04

<pre><code>
Mage2Docker version: v3.4.0 ( 6ad018 ) 
Magento version: 2.4.1 

 
Others:

</code></pre>

**Comment by damienwebdev** [comment]
@nwtben [Magento 2](https://github.com/magento/community-features/issues/302) won't be supporting composer 2 for a while, so I will be pinning composer to version 1. 

I'll get a release out ASAP today, thanks for the report!

**Comment by damienwebdev** [comment]
@nwtben I've released this with v3.5.0, please `docker pull` to get the updated tag, or simply just use the:
 
`magento-php:7.X-fpm-alpine-v3.5.0` image.
